adapters/adview/adview.go

```diff
@@ -157,3 +178,44 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 	}
 	return mediaType, nil
 }
+
+func getMediaTypeForBid(bid openrtb2.Bid, reqImps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range reqImps {
+		if bid.ImpID == imp.ID {
+			if bid.Ext != nil {
+				var bidExt *adviewBidExt
+				err := json.Unmarshal(bid.Ext, &bidExt)
+				if err == nil {
+					return getBidType(bidExt), nil
+				} else {
+					return mediaType, err
+				}
```

Comment:
code path is not covered. should add json test for better coverage

Reply:
thx, and I add a test script banner-app-resp-no-formattype.json, and also adview ad server always deliver 'formattype' in ad response, so formattype error cases may not be occurred.